tests/0094-idempotence_msg_timeout.c

```diff
@@ -213,6 +215,114 @@ static void do_test_produce_timeout(const char *topic, const int msgrate) {
                  msgrate);
 }
 
+
+static void do_test_epoch_exhaustion(const char *topic) {
+        rd_kafka_t *rk;
+        rd_kafka_topic_t *rkt;
+        rd_kafka_conf_t *conf;
+        rd_kafka_pid_t pid, new_pid;
+        rd_kafka_resp_err_t err;
+        sockem_ctrl_t ctrl;
+        uint64_t testid;
+        char buf[1024];
+        char key[128];
+        int msgcnt          = 20;
+        const int partition = RD_KAFKA_PARTITION_UA;
+
+        SUB_TEST("Test idempotent producer epoch exhaustion");
+
+        testid = test_id_generate();
+
+        test_conf_init(&conf, NULL, 60);
+        sockem_ctrl_init(&ctrl);
+
+        test_conf_set(conf, "enable.idempotence", "true");
+        test_conf_set(conf, "batch.size", "1");
+        test_conf_set(conf, "linger.ms", "1");
+        test_conf_set(conf, "retries", "1");
+
+        rd_kafka_conf_set_dr_msg_cb(conf, my_dr_msg_cb);
+
+        test_socket_enable(conf);
+        test_curr->is_fatal_cb = is_fatal_cb;
+
+        rk  = test_create_handle(RD_KAFKA_PRODUCER, conf);
+        rkt = test_create_producer_topic(rk, topic, "message.timeout.ms", "100",
+                                         NULL);
+
+        /* Create the topic to make sure connections are up and ready. */
+        err = test_auto_create_topic_rkt(rk, rkt, tmout_multip(5000));
+        TEST_ASSERT(!err, "topic creation failed: %s", rd_kafka_err2str(err));
+
+        if (rd_kafka_test_idemp_get_pid(rk, &pid.id, &pid.epoch) != 0) {
```

Comment:
With the new error signature this can be simplified by:
`TEST_CALL_ERR__(rd_kafka_test_idemp..(...));` which automatically checks for error and asserts accordingly